aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -114,8 +110,14 @@ Completable hydrate() {
         TopologicalOrdering ordering =
             TopologicalOrdering.forRegisteredModels(schemaRegistry, modelProvider);
         Collections.sort(modelSchemas, ordering::compare);
+        ArrayList<String> toBeSyncedModelArray = new ArrayList<>();
         for (ModelSchema schema : modelSchemas) {
-            hydrationTasks.add(createHydrationTask(schema));
+            //Check to see if query predicate for this schema is not equal to none. This means customer does
+            // not want to sync the data for this model.
+            if (!QueryPredicates.none().equals(queryPredicateProvider.getPredicate(schema.getName()))) {
+                hydrationTasks.add(createHydrationTask(schema));
+                toBeSyncedModelArray.add(schema.getName());
+            }
```

Comment:
just a bit confusing comment paired with the code, i think it should be line up, like:
```
//Check to see if query predicate for this schema is equal to none. This means customer does
            // not want to sync the data for this model.
if (QueryPredicates.none().equals(queryPredicateProvider.getPredicate(schema.getName()))) {
   continue;
}

hydrationTasks.add(createHydrationTask(schema));
toBeSyncedModelArray.add(schema.getName());
```